### PR TITLE
Feature/mix design proportion

### PR DESCRIPTION
**Pull request by sujanthan02**

### CODE REVIEW

File: tokyo-supermix-data-common/src/main/java/com/tokyo/supermix/data/dto/MixDesignProportionRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+public class MixDesignProportionRequestDto {
+
+  private Long id;
+   @NotNull(message = "{mixDesignProportionRequestDto.quantity.null}")
+   @NotEmpty(message = "{mixDesignProportionRequestDto.quantity.empty}")
+  private Long quantity;
+  private String unitId;
+  private String mixDesignCode;
+  private String rawMaterialId;
```
Comment (Keerthana65) [inline tokyo-supermix-data-common/src/main/java/com/tokyo/supermix/data/dto/MixDesignProportionRequestDto.java]: Change the data type: private Long rawMaterialId;

File: tokyo-supermix-data-common/src/main/java/com/tokyo/supermix/data/dto/MixDesignProportionResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package com.tokyo.supermix.data.dto;
+
+public class MixDesignProportionResponseDto {
+  private Long id;
+  private Long quantity;
+  private String unitId;
+  private String unitUnit;
```
Comment (Keerthana65) [inline tokyo-supermix-data-common/src/main/java/com/tokyo/supermix/data/dto/MixDesignProportionResponseDto.java]: Change the variable name: private String unit;

File: tokyo-supermix-data-common/src/main/java/com/tokyo/supermix/data/dto/MixDesignProportionResponseDto.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  private String mixDesignCode;
+  private String rawMaterialId;
```
Comment (Keerthana65) [inline tokyo-supermix-data-common/src/main/java/com/tokyo/supermix/data/dto/MixDesignProportionResponseDto.java]: Change the data type: private Long rawMaterialId;

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/EndpointURI.java [hunk 4]
```diff
@@ -187,6 +187,16 @@
   public static final String MIXDESIGNS = BASE_API_PATH + "mixdesigns";
   public static final String GET_MIXDESIGN_BY_ID = MIXDESIGN + CODE;
   public static final String DELETE_MIXDESIGN_BY_ID = MIXDESIGN + CODE;
+  
+  /*
+   * Mix design propation apis
+   * 
+   */
+  
+  public static final String MIXDESIGN_PROPORTION = BASE_API_PATH + "mixdesign-proporation";
```
Comment (Keerthana65) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/EndpointURI.java]: Change the spelling mistake: mix-design-proportion

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/services/MixDesignProportionServiceImpl.java [hunk 5]
```diff
@@ -1,8 +1,57 @@
...
 
 @Service
 public class MixDesignProportionServiceImpl implements MixDesignProportionService {
 
+    @Autowired
+    public MixDesignProportionRepository mixDesignProportionRepository;
+
+    @Transactional(readOnly = true)
+    public List<MixDesignProportion> getAllMixDesignProportions() {
+
```
Comment (Keerthana65) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/services/MixDesignProportionServiceImpl.java]: Removed line space & add google style

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/MixDesignProportionController.java [hunk 6]
```diff
@@ -1,9 +1,125 @@
...
+        mixDesignProportionRequestDtoObj -> mixDesignProportionService.saveMixDesignProportion(
+            mapper.map(mixDesignProportionRequestDtoObj, MixDesignProportion.class)));
+    return new ResponseEntity<Object>(
+        new BasicResponse<>(RestApiResponseStatus.OK, Constants.ADD_MIXDESIGN_PROPORTION_SUCCESS),
+        HttpStatus.OK);
+  }
+
+  @GetMapping(value = EndpointURI.MIXDESIGN_PROPORTIONS)
+  public ResponseEntity<Object> getAllMixDesignProportions() {
+    List<MixDesignProportion> mixDesignProportionList =
```
Comment (Keerthana65) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/MixDesignProportionController.java]: avoid the creating object, here you can directly pass. for example mapper.map( mixDesignProportionService.getAllMixDesignProportions(), MixDesignProportionResponseDto.class),

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/MixDesignProportionController.java [hunk 7]
```diff
@@ -1,9 +1,125 @@
...
+          HttpStatus.OK);
+    }
+    return new ResponseEntity<>(
+        new ValidationFailureResponse(Constants.MIXDESIGN_PROPORTION,
+            validationFailureStatusCodes.getMixDesignProportionAlreadyExist()),
+        HttpStatus.BAD_REQUEST);
+
+  }
+
+  @GetMapping(value = EndpointURI.MIXDESIGN_PROPORTION_BY_MIXDESIGNCODE)
```
Comment (ramz7) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/MixDesignProportionController.java]: give error response

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/MixDesignProportionController.java [hunk 8]
```diff
@@ -1,9 +1,125 @@
...
+  public ResponseEntity<Object> getMixDesignProportionByMixDesignCode(
+      @PathVariable String mixDesignCode) {
+    logger.debug("Get mix design proportion by id ");
+    return new ResponseEntity<>(
+        new ContentResponse<>(Constants.MIXDESIGN_PROPORTION_ID,
+            mixDesignProportionService.findByMixDesign(
+                mixDesignService.getMixDesignById(mixDesignCode)),
```
Comment (ramz7) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/MixDesignProportionController.java]: keep method name as getMixDesignByCode, (mix Design has code)

Comment (ramz7) [reply tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/MixDesignProportionController.java]: if use findByMixDesignCode, no need to use mixDesignService here..

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/EndpointURI.java [hunk 9]
```diff
@@ -187,6 +187,16 @@
...
+  public static final String MIXDESIGN_PROPORTIONS = BASE_API_PATH + "mixdesign-proporations";
```
Comment (Mathuragini) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/EndpointURI.java]: consider the spell mistake in all the code sectors- mix-design-proportion

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/MixDesignProportionController.java [hunk 10]
```diff
@@ -1,9 +1,125 @@
...
+    if (mixDesignProportionService.isMixDesignProportionExist(id)) {
+      mixDesignProportionService.deleteById(id);
+      return new ResponseEntity<Object>(
+          new BasicResponse<>(RestApiResponseStatus.OK, Constants.MIXDESIGN_PROPORTION_DELETED),
+          HttpStatus.OK);
+    }
+    logger.debug("Invalid Id");
+    return new ResponseEntity<>(
+        new ValidationFailureResponse(Constants.MIXDESIGN_DELETED,
+            validationFailureStatusCodes.getMixDesignProportionAlreadyExist()),
```
Comment (Mathuragini) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/MixDesignProportionController.java]: How can you pass the constant message as MIXDESIGN_DELETED at here? its is else part.

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/MixDesignProportionController.java [hunk 11]
```diff
@@ -1,9 +1,125 @@
...
+  public ResponseEntity<Object> getMixDesignProportionById(@PathVariable Long id) {
+    if (mixDesignProportionService.isMixDesignProportionExist(id)) {
+      logger.debug("Get mix design proportion by id ");
+      return new ResponseEntity<>(new ContentResponse<>(Constants.MIXDESIGN_PROPORTION_ID,
+          mapper.map(mixDesignProportionService.getMixDesignProportionById(id),
+              MixDesignProportionResponseDto.class),
+          RestApiResponseStatus.OK), HttpStatus.OK);
+    }
+    logger.debug("Invalid Id");
+    return new ResponseEntity<>(new ValidationFailureResponse(Constants.MIXDESIGN_PROPORTION_ID,
```
Comment (Mathuragini) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/MixDesignProportionController.java]: Carefully consider the responses and validation message passing. for the validation failure response, you have to pass the MIXDESIGN in constants.

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/services/MixDesignProportionService.java [hunk 12]
```diff
@@ -1,5 +1,16 @@
...
+import com.tokyo.supermix.data.entities.MixDesign;
+import com.tokyo.supermix.data.entities.MixDesignProportion;
 
+public interface MixDesignProportionService {
+  public List<MixDesignProportion> getAllMixDesignProportions();
+  public MixDesignProportion saveMixDesignProportion(MixDesignProportion mixDesignProportion);
+  public void deleteById(Long id);
+  public MixDesignProportion getMixDesignProportionById(Long id);
+  public boolean isIdExist(Long id);
+  public boolean isMixDesignProportionExist(Long id);
```
Comment (Mathuragini) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/services/MixDesignProportionService.java]: what is the difference between those 2 services? Both do same functionalities.

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/util/ValidationFailureStatusCodes.java [hunk 13]
```diff
@@ -1079,5 +1079,55 @@ public void setProjectAlreadyExist(String projectAlreadyExist) {
...
+
+  @Value("${validation.mixDesignProportion.alreadyExist}")
+  private String mixDesignProportionAlreadyExist;
+  
+  
+  @Value("${mixDesignProportionRequestDto.quantity.null}")
+  private String quantityNotNull;
+
+  @Value("${mixDesignProportionRequestDto.quantity.empty}")
+  private String quantityNotEmpty;
```
Comment (Mathuragini) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/util/ValidationFailureStatusCodes.java]: use standard naming conventions, quantityIsNull

**Pull request merged**